SunViewPlayground/MyPlayground.playground/Contents.swift

```diff
@@ -152,27 +152,27 @@ class SunView: UIView {
         return SunViewLocation(x: absoluteX, y: absoluteY)
     }
 
-    func updateInterface(controller: DaylightModelController) {
-        self.superview?.backgroundColor = controller.primaryColor
-        self.backgroundColor = controller.primaryColor
-        self.sunriseLabel.textColor = controller.secondaryColor
-        self.sunsetLabel.textColor = controller.secondaryColor
-        self.dayTimeLabel.textColor = controller.secondaryColor
-        self.nightTimeLabel.textColor = controller.secondaryColor
-        self.horizon.backgroundColor = controller.secondaryColor
-        self.sun.tintColor = controller.secondaryColor
-        self.moon.tintColor = controller.secondaryColor
-
-        self.sunViewLocation = self.location(for: controller.percentageInDay)
-        self.moon.isHidden = !controller.shouldShowMoon
-        self.nightTimeLabel.isHidden = !controller.shouldShowMoon
-        self.sun.isHidden = controller.shouldShowMoon
-        self.dayTimeLabel.isHidden = !controller.shouldShowTimeLabel
-
-        self.dayTimeLabel.text = controller.currentTimeString
-        self.nightTimeLabel.text = controller.currentTimeString
-        self.sunriseLabel.text = controller.sunriseTimeString
-        self.sunsetLabel.text = controller.sunsetTimeString
+    func updateInterface(daylightModelController: DaylightModelController) {
```

Comment:
i think here you could use `with` as the external parameter as well. 

Also, for all these - I don't think you necessarily have to have the `model` bit in the name of the local var. I think calling it `daylightController` helps distinguish what controller it is, and the type shows that it's a model controller. You can leave it if you think it makes things clearer though.

Reply:
cool i will do that ✌️